.github/workflows/test_nrf9160dk.yml

```diff
@@ -117,7 +117,9 @@ jobs:
         run: |
           cd samples/test
           source $HOME/runner_env.sh
-          nrfjprog --program build/zephyr/merged.hex --sectorerase --verify -f NRF91 --snr $CI_NRF9160DK_SNR
+          nrfjprog --eraseall -f NRF91 --snr $CI_NRF9160DK_SNR
+          nrfjprog --program build/zephyr/merged.hex --verify -f NRF91 --snr $CI_NRF9160DK_SNR
           nrfjprog --pinreset -f NRF91 --snr $CI_NRF9160DK_SNR
+          sleep 3
```

Comment:
```
* Add delay between reset and verification testing
```
this answers WHAT has changed, but there is missing WHY information in commit message. An example of error message would be great (which appears without this delay).

Reply:
Same for `* Erase and program flash in separate steps`...

Also, will this be applicable for `nrf52840dk_nrf52840` platform? i.e. https://github.com/golioth/golioth-zephyr-sdk/actions/runs/4690358408/jobs/8313466681?pr=362 ?

Reply:
I updated the commit message. The error output now throws checkpatch and gitlint errors. :shrug: 

> Also, will this be applicable for nrf52840dk_nrf52840 platform? i.e. https://github.com/golioth/golioth-zephyr-sdk/actions/runs/4690358408/jobs/8313466681?pr=362 ?

Yes, this is my theory. We had discussed adding `nrfjprog --recover` but my hope is a separate erase step would both avoid this issue, and correct it if it does occur.